pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -242,3 +248,31 @@ func panicHandler(handler http.Handler) http.Handler {
 		handler.ServeHTTP(w, r)
 	})
 }
+
+func setDefaultMiddleware(config *configuration.Configuration) (changed bool, err error) {
+	// Default to openshift middleware for relevant types
+	// This allows custom configs based on old default configs to continue to work
+	if config.Middleware == nil {
+		config.Middleware = map[string][]configuration.Middleware{}
+	}
+	for _, middlewareType := range []string{"registry", "repository", "storage"} {
+		if _, exists := config.Middleware[middlewareType]; !exists {
```

Comment:
if you're going to search and append anyway, you can drop this check... append will allocate it if needed